src/drivers/spi/spi.c

```diff
@@ -0,0 +1,175 @@
+/**
+ * @file spi.c
+ * @brief
+ * @author Denis Deryugin <deryugin.denis@gmail.com>
+ * @version
+ * @date 03.12.2018
+ */
+
+#include <drivers/spi.h>
+#include <embox/unit.h>
+#include <fs/dvfs.h>
+#include <util/log.h>
+
+#include <assert.h>
+#include <errno.h>
+#include <sys/uio.h>
+
+ARRAY_SPREAD_DECLARE(const struct dev_module, __char_device_registry);
+
+static int spi_init(void) {
+	struct dev_module *dev_module;
+	struct spi_device *sdev;
+
+	/* Note: this currently is neccessary only because
+	 * common probe() is not implemented for serial devices */
+	array_spread_foreach_ptr(dev_module, __char_device_registry) {
+		if (dev_module->device->dev_iops != &spi_iops) {
```

Comment:
So SPI now depends on dvfs in any way?

Reply:
Yes, it depends on devfs.

Probably, separate modules can be used for devfs-related stuff and general SPI functions. But this leads SPI devices be **not** a specific case for char devices (i.e. we should duplicate char_dev pool with spi_dev pool and so on).

Look at the last commit, I splitted source into two files. If it will be just two modules, then for non-devfs module we will have no interface to get a device in any way. I'm not sure if it's ok.

Reply:
But here's another question: are there any configs which do not use FS at all?

Reply:
Since we didn't use SPI well yet (only 1-2 platforms), it's hard to say... But anyway, after your last commit it looks better to me.

Reply:
Check out new version for SPI module, please

Reply:
Now it looks good to me :)